src/Gaffer/Animation.cpp

```diff
@@ -1925,36 +1987,86 @@ Animation::KeyPtr Animation::CurvePlug::insertKeyInternal( const float time, con
 		return key;
 	}
 
-	// get interpolator
+	// get interpolator and tie mode
 
 	const Interpolator* interpolator = Interpolator::getFactory().getDefault();
+	TieMode tieMode = defaultTieMode();
 
 	if( lo )
 	{
 		interpolator = lo->m_interpolator;
+		tieMode = lo->m_tieMode;
 	}
 	else if( const Key* const kf = firstKey() )
 	{
 		interpolator = kf->m_interpolator;
+		tieMode = kf->m_tieMode;
 	}
 
 	assert( interpolator );
 
 	// create key
 
-	key.reset( new Key( time, ( ( value != nullptr ) ? ( *value ) : 0.f ), interpolator->getInterpolation() ) );
+	key.reset( new Key( time, ( ( value != nullptr ) ? ( *value ) : 0.f ), interpolator->getInterpolation(),
+		defaultSlope(), defaultScale(), defaultSlope(), defaultScale(), TieMode::Manual ) );
 
 	// if specified value is the same as the evaluated value of the curve then bisect span.
 
 	if( ( lo && hi ) && ( ( value == nullptr ) || ( evaluate( time ) == ( *value ) ) ) )
```

Comment:
This is existing code, but when would the `evaluate( time ) == ( *value )` ever actually trigger?  If it's actually a complex curve where bisection would produce a different result to auto slope, then using exact comparison of floating point values feels like it could have precision concerns ... but maybe we never use this case anyway?  It seems a lot better to not specify a value when you want this behaviour.

Reply:
this case triggers from the numeric plug editor widget in the node editor when setting a key via the context menu, the value in the editor is obtained by calling evaluate( time ) so should match exactly. for all other values, ie. when the user has entered something different, we currently do not do a bisection.